contracts/staking/GoodFundManager.sol

```diff
@@ -225,15 +226,17 @@ contract GoodFundManager is DAOContract {
 
 			if (i == 0) break; // when active contracts length is 1 then gives error
 		}
-
-		// Finds the actual transferred iToken
-		uint256 interest = iToken.balanceOf(reserveAddress) - currentBalance;
-
+		// Finds the actual transferred DAI
+		uint256 interest = daiToken.balanceOf(reserveAddress) - currentBalance;
+		// Convert DAI to cDAI and continue further transactions with cDAI
+		uint256 cDAIBalance = iToken.balanceOf(reserveAddress);
```

Comment:
we can make a little gas optimization here
record cDAIbalance before the iteration, this way we can have DAI staking simply transfer to the reserve cDAI and not DAI.
then modify compoundstaking to simply transfer cDAI in case this is the iToken

Reply:
I modified like that @sirpy